src/Datadog.Trace/HttpOverStreams/DatadogHttpClient.cs

```diff
@@ -0,0 +1,100 @@
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Datadog.Trace.HttpOverStreams.HttpContent;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.HttpOverStreams
+{
+    internal class DatadogHttpClient
+    {
+        private const int BufferSize = 10240;
+        private static readonly Vendors.Serilog.ILogger Logger = DatadogLogging.For<DatadogHttpClient>();
+
+        public Task<HttpResponse> SendAsync(HttpRequest request, Stream requestStream, Stream responseStream, CancellationToken cancellationToken)
+        {
+            Task.Run(() => SendRequest(request, requestStream, cancellationToken), cancellationToken);
```

Comment:
Why are you sending the request and reading the response in parallel?

Reply:
Removed this